sycl/source/detail/graph_impl.cpp

```diff
@@ -1435,94 +1491,17 @@ void exec_graph_impl::update(
   // ensure it is ordered correctly.
   NeedScheduledUpdate |= MExecutionEvents.size() > 0;
```

Comment:
Can't comment on the line directly but we're leaving execution event cleanup in `needsScheduledUpdate()` which is a little strange to me since it does not affect the bool returned from this function, would prefer to keep that in the main update function so it's clear what is happening.